File: frontend/types/admin-geral.ts
export type AdminAccess = {
  admin_geral: boolean;
  usuario_id?: string;
  nome?: string;
  email?: string;
};

export type AdminResumo = {
  familias_total: number;
  familias_ativas: number;
  familias_suspensas: number;
  familias_novas_30_dias: number;
  usuarios_total: number;
  membros_ativos: number;
  compras_total: number;
  itens_total: number;
  produtos_total: number;
  supermercados_total: number;
  administradores_sistema: number;
  gerado_em: string;
};

export type AdminFamiliaResumo = {
  id: string;
  nome: string;
  status: string;
  observacao_admin: string;
  criado_em: string;
  atualizado_em: string;
  ultima_atividade?: string | null;
  membros_count: number;
  administradores_count: number;
  compras_count: number;
  produtos_count: number;
  itens_count: number;
  administrador_nome: string;
  administrador_email: string;
};

export type AdminMembro = {
  usuario_id: string;
  nome: string;
  email: string;
  papel: string;
  status: string;
  criado_em: string;
  familia_atual: boolean;
  admin_geral: boolean;
};

export type AdminFamiliaDetalhes = {
  familia: AdminFamiliaResumo & {
    supermercados_count: number;
    suspensa_em?: string | null;
  };
  membros: AdminMembro[];
};

export type AdminUsuarioResumo = {
  id: string;
  nome: string;
  email: string;
  criado_em: string;
  atualizado_em: string;
  familia_atual_id?: string | null;
  familia_atual_nome: string;
  familia_atual_status: string;
  admin_geral: boolean;
  familias_count: number;
  ultima_atividade?: string | null;
};

export type AdminUsuarioFamilia = {
  familia_id: string;
  familia_nome: string;
  familia_status: string;
  papel: string;
  status: string;
  criado_em: string;
  familia_atual: boolean;
};

export type AdminUsuarioDetalhes = {
  usuario: AdminUsuarioResumo;
  familias: AdminUsuarioFamilia[];
};

export type AdminAuditoriaRegistro = {
  id: string;
  acao: string;
  entidade: string;
  entidade_id?: string | null;
  familia_id?: string | null;
  usuario_alvo_id?: string | null;
  resumo: string;
  dados_anteriores: Record<string, unknown>;
  dados_novos: Record<string, unknown>;
  origem_ip?: string | null;
  request_id?: string | null;
  criado_em: string;
  administrador_nome: string;
  administrador_email: string;
};

export type Paginated<T, K extends string> = {
  total: number;
  limite: number;
  offset: number;
  tem_mais: boolean;
} & Record<K, T[]>;

export type AdminFamiliasLista = Paginated<AdminFamiliaResumo, "familias">;
export type AdminUsuariosLista = Paginated<AdminUsuarioResumo, "usuarios">;
export type AdminAuditoriaLista = Paginated<AdminAuditoriaRegistro, "registros">;

export type AdminAction = {
  mensagem: string;
  familia_id?: string;
  usuario_id?: string;
  status?: string;
  papel?: string;
  nome?: string;
  observacao_admin?: string;
  familia_nome?: string;
  usuarios_sem_familia?: number;
  proxima_familia_id?: string | null;
};

File: backend/app/schemas/admin_geral.py
from typing import Any, Literal

from pydantic import BaseModel, Field


class AdminAccessResponse(BaseModel):
    admin_geral: bool
    usuario_id: str | None = None
    nome: str | None = None
    email: str | None = None


class AdminResumoResponse(BaseModel):
    familias_total: int = Field(ge=0)
    familias_ativas: int = Field(ge=0)
    familias_suspensas: int = Field(ge=0)
    familias_novas_30_dias: int = Field(ge=0)
    usuarios_total: int = Field(ge=0)
    membros_ativos: int = Field(ge=0)
    compras_total: int = Field(ge=0)
    itens_total: int = Field(ge=0)
    produtos_total: int = Field(ge=0)
    supermercados_total: int = Field(ge=0)
    administradores_sistema: int = Field(ge=0)
    gerado_em: str
    administrador_id: str


class AdminListResponse(BaseModel):
    total: int = Field(ge=0)
    limite: int = Field(ge=1)
    offset: int = Field(ge=0)
    tem_mais: bool
    administrador_id: str


class AdminFamiliasResponse(AdminListResponse):
    familias: list[dict[str, Any]]


class AdminUsuariosResponse(AdminListResponse):
    usuarios: list[dict[str, Any]]


class AdminAuditoriaResponse(AdminListResponse):
    registros: list[dict[str, Any]]


class AdminFamiliaDetalhesResponse(BaseModel):
    familia: dict[str, Any]
    membros: list[dict[str, Any]]
    administrador_id: str


class AdminUsuarioDetalhesResponse(BaseModel):
    usuario: dict[str, Any]
    familias: list[dict[str, Any]]
    administrador_id: str


class AtualizarFamiliaAdminRequest(BaseModel):
    nome: str | None = Field(default=None, min_length=2, max_length=80)
    observacao: str | None = Field(default=None, max_length=1000)


class AlterarStatusFamiliaAdminRequest(BaseModel):
    motivo: str = Field(min_length=3, max_length=500)


class ExcluirFamiliaAdminRequest(BaseModel):
    nome_confirmacao: str = Field(min_length=2, max_length=80)
    confirmacao: Literal["EXCLUIR DEFINITIVAMENTE"]
    motivo: str = Field(min_length=3, max_length=500)


class AlterarPapelMembroAdminRequest(BaseModel):
    papel: Literal["administrador", "membro"]


class ExcluirUsuarioAdminRequest(BaseModel):
    email_confirmacao: str = Field(min_length=5, max_length=254)
    confirmacao: Literal["EXCLUIR DEFINITIVAMENTE"]
    motivo: str = Field(min_length=3, max_length=500)


class AdminActionResponse(BaseModel):
    mensagem: str
    familia_id: str | None = None
    usuario_id: str | None = None
    status: str | None = None
    papel: str | None = None
    nome: str | None = None
    observacao_admin: str | None = None
    familia_nome: str | None = None
    usuarios_sem_familia: int | None = None
    resumo_excluido: dict[str, Any] | None = None
    proxima_familia_id: str | None = None
